lib/spack/docs/binary_caches.rst

```diff
@@ -0,0 +1,99 @@
+.. _binary_caches:
+
+Build caches
+============================
+
+Some sites may encourage users to set up their own test environments
+before carrying out central installations, or some users prefer to set
+up these environments on their own motivation. To reduce the load of
+recompiling otherwise identical package specs in different installations,
+created build artifacts can be put into build cache tarballs, uploaded to 
+your spack mirror and then downloaded and installed by others.
+
+
+Creating build cache files
+--------------------------
+
+A compressed tarball of an installed package is created.
+Tarballs are created for all of its link and run 
+dependency packages as well.
+Compressed tarballs are signed with gpg and 
+signature and tarball and put in a ".spack" tarfile. 
+Optionally the rpaths ( and ids and deps on macOS )
+can be changed to paths relative to the spack install tree 
+before the tarball is created.
+
+Build caches are created via:
+
+.. code-block:: sh
+
+   $ spack buildcache create 
+
+
+Finding and installing build cache files
+------------------------------------
+
+To find install build caches a spack mirror must be configured with
+ 
+``spack mirror add <name> <url>``. 
+
+Build caches are found via: 
+
+.. code-block:: sh
+
+   $ spack buildcache list
+
+Build caches are installed via:
+
+.. code-block:: sh
+
+   $ spack buildcache install 
+   
+
+Relocation
+-------------------------------
+
+Initial build and later installation do not necessarily happen at the same
+location. Spack provides a very relocation capability and corrects for
+RPATHs and non-relocatable scripts. However, many packages compile paths into
+binary artificats directly. In such cases, the build instructions of this
+package would need to be adjusted for better re-locatability.
+
+
+Usage 
+-------------------------------
+spack buildcache create <>
+
+options:
+
+-d : directory to create build_cache direcory in, 
```

Comment:
This looks like it could use some markup. Also, `build_cache` is not an (English) word :) .